hugo: Add example of anchor link usage

JorindeUsMedia opened the pull request:
- Add example of anchor link using ref to the example/ref page
- Change links in column to '/' instead of a non-existing anchor link

Closes https://github.com/cue-lang/cue/issues/2971

jpluscplusm commented on content/examples/shortcodes/ref-relref/en.md:
```diff
@@ -23,3 +24,20 @@ The result will be just like a link. But the difference is that with ref/relref
 
 [Buttons example]({{< relref "buttons/index.md" >}})
 
+Relref can also be used for anchor links:
```
If I'm interpretting this change correctly, please add a note that same-page links to a fragment "*must*" be invoked via `ref`/`relref`, if they're to work consistently.

jpluscplusm commented on content/examples/shortcodes/ref-relref/en.md:
````diff
@@ -23,3 +24,20 @@ The result will be just like a link. But the difference is that with ref/relref
+Relref can also be used for anchor links:
+
+```
+[Anchor links example]({{<ref "#anchor-header" >}})
````
This example renders oddly, negating the point that I think you're trying to make (that `relref` *must* be used, and **not** just bare fragments inside a Markdown link):

![image](https://github.com/user-attachments/assets/dcbce452-3828-4fd1-be71-df7ce8b63de6)

jpluscplusm reviewed (changes requested):
@JorindeUsMedia I don't think this PR leaves us in a state where either https://github.com/cue-lang/cue/issues/2971 is fixed; or where we understand what we have to do to avoid it; and where we have on-page examples of things working "correctly" to review and cross-check in the future.

https://github.com/cue-lang/cue/issues/2971 is solely concerned with links to fragments on the same page, when placed inside columns or info/warning/caution blocks. If this issue is demonstrating a workaround for that problem, please can we place some examples on page that show it being fixed (with the use of `relref`?).

If this issue **isn't** demonstrating that, then I'm not sure of its purpose, and I don't think it should close https://github.com/cue-lang/cue/issues/2971.

JorindeUsMedia commented:
@jpluscplusm It seems Hugo adds the baseUrl in front of all links: also anchor links. 
See https://discourse.gohugo.io/t/link-with-anchor-has-leading-slash/30391
They suggest to use relref instead. I did some more googling but could not find a solution for hugo/goldmark not adding the baseUrl to anchor links.

JorindeUsMedia commented:
@jpluscplusm what do I need to do in order to get this pr closed?  Mention that is must be used on the examples page?

jpluscplusm commented:
> @jpluscplusm what do I need to do in order to get this pr closed? Mention that is must be used on the examples page?

I think an explicit mention on the example page that linking to an anchor on the same page requires `relref`, with an example of it being used successfully inside the set of shortcodes that are mentioned in the linked issue: `columns` and `info`, at a minimum.

JorindeUsMedia commented:
Hm it seems even with using ref instead of regular links there still seems to be an issue with Hugo (or Goldmark) adding the baseUrl in front of the anchor links.. I'll have to dive into the issue again.

JorindeUsMedia commented:
I think I found it:

https://gohugo.io/functions/transform/markdownify/ > 

> Although the markdownify function honors [Markdown render hooks](https://gohugo.io/render-hooks/) when rendering Markdown to HTML, use the RenderString method instead of markdownify if a render hook accesses .Page context. See issue [#9692](https://github.com/gohugoio/hugo/issues/9692) for details.

https://github.com/gohugoio/hugo/issues/9692

I think we need to use 

{{- $.Page.RenderString .Inner  -}}

jpluscplusm commented:
Hey @JorindeUsMedia - I've just hit this issue when trying to link to a same-page fragment from inside a `caution` block. Is there a workaround that works with the current site, other than hardcoding a link to the absolute path of the current page?

JorindeUsMedia commented:
> Hey @JorindeUsMedia - I've just hit this issue when trying to link to a same-page fragment from inside a `caution` block. Is there a workaround that works with the current site, other than hardcoding a link to the absolute path of the current page?

I've found a solution: see my last comment. I also implemented it but didn't have time to test it yet. I've just updated the pr

jpluscplusm commented:
I can see changes to several different elements that we use throughout the site.

I'm just writing this as a marker that I've seen this PR, but need to find a bit of time to sit down and compare some before-and-after uses of the elements that have been updated :-)

jpluscplusm commented:
I've started CI manually, [here](https://review.gerrithub.io/c/cue-lang/cuelang.org/+/1203530), as a change we made recently has made the Netlify preview bot unhappy. This is a just a temporary work around until we fix the bot.

jpluscplusm reviewed (changes requested):
Looks generally good - just a few questions/etc.

I'd like to grab a brief chat to understand the core of what's changing here, so that I can make sure we're marking up anchor links correctly.

JorindeUsMedia commented on content/examples/shortcodes/spotlight/en.md:
```diff
@@ -12,8 +12,8 @@ In order to spotlight some content, you can use `{{</* spotlight */>}}`.
 
-- **A tool to edit text files.** Any text editor you have will be fine. Some popular text editors are VSCode, Vim, Nano, and Emacs; but if you need to use Notepad that will work fine.
-- **A command terminal.** `cue` works well using any terminal on Linux and Mac, and on PowerShell or `cmd.exe` in Windows.
+- **A tool to edit text files.** Any text editor you have will be fine. Some popular text editors are VSCode, Vim, Nano, and Emacs; but if you need to use Notepad that will work fine.
```
previous character: NBSP (non breakable space)
changed to: regular space

JorindeUsMedia commented on content/examples/shortcodes/spotlight/en.md:
```diff
@@ -24,8 +24,8 @@ The rendered output looks like this:
--   **A tool to edit text files.** Any text editor you have will be fine. Some popular text editors are VSCode, Vim, Nano, and Emacs; but if you need to use Notepad that will work fine.
--   **A command terminal.** `cue` works well using any terminal on Linux and Mac, and on PowerShell or `cmd.exe` in Windows.
+-   **A tool to edit text files.** Any text editor you have will be fine. Some popular text editors are VSCode, Vim, Nano, and Emacs; but if you need to use Notepad that will work fine.
+-   **A command terminal.** `cue` works well using any terminal on Linux and Mac, and on PowerShell or `cmd.exe` in Windows.
```
same here

![image](https://github.com/user-attachments/assets/a3269f75-19b6-4bd0-adbc-88745c1814ce)

JorindeUsMedia commented on content/examples/shortcodes/ref-relref/en.md:
```diff
@@ -15,6 +15,7 @@ Read a more extensive description of `ref` and `relref` in the [Hugo cross refer
 [Links example]({{</* ref "examples/basic/text/index.md#links" */>}})
 
 [Buttons example]({{</* relref "buttons/index.md" */>}})
+
```
yes unintentional: removed the new line

JorindeUsMedia commented on content/examples/shortcodes/columns/en.md:
```diff
@@ -31,13 +31,13 @@ turpis id tristique blandit, purus orci efficitur velit, nec [molestie tortor es
 Lorem ipsum *dolor sit amet*, consectetur adipiscing elit. Phasellus tristique lectus ut bibendum tincidunt. Vestibulum posuere nisl quis bibendum fermentum.
 Aenean viverra commodo erat ac porttitor. Suspendisse **a interdum leo**. Etiam volutpat lectus auctor, porttitor urna et, egestas quam. Curabitur sodales,
-turpis id tristique blandit, purus orci efficitur velit, nec [molestie tortor est eget](#link) erat.
+turpis id tristique blandit, purus orci efficitur velit, nec [molestie tortor est eget](/) erat.
```
`#link` didn't go anywhere so I changed it to '/'. After our previous conversation I've now removed the link.

jpluscplusm commented:
Preview available at https://cl-1203530-2--cue-cls.netlify.app/.

@JorindeUsMedia I've resolved all the threads I feel able to - please could you resolve the rest, and then I'll know it's good for final review? Thank you!

JorindeUsMedia commented on content/examples/shortcodes/ref-relref/en.md:
```diff
@@ -23,3 +24,20 @@ The result will be just like a link. But the difference is that with ref/relref
 
 [Buttons example]({{< relref "buttons/index.md" >}})
 
+Ref can also be used for anchor links:
```
I think we answered this in our meeting last week. It's ok to use them as you do now

JorindeUsMedia commented on content/examples/basic/text/en.md:
```diff
@@ -34,22 +34,23 @@ Also on simple detail page's you should generally not use a Heading 1 in the con
+### Heading - Custom Ids {#custom-id}
 
-Adding custom IDs allows you to link directly to headings. To add a custom heading ID, enclose the custom ID in curly braces on the same line as the heading.
+ By default Hugo adds the slugified header as the id. For instance: "Heading 2" becomes id 'heading-2'.  Adding custom IDs allows you to link directly to headings using a custom id/anchor href.
```
I've added your suggestion.

JorindeUsMedia commented:
> @JorindeUsMedia I've resolved all the threads I feel able to - please could you resolve the rest, and then I'll know it's good for final review? Thank you!

All resolved :)

jpluscplusm commented on hugo/layouts/shortcodes/logos__item.html:
```diff
@@ -78,7 +78,7 @@
             <div class="dialog__content">
                 {{ with .Inner }}
-                    {{ . | markdownify }}
+                    {{ . | $.Page.RenderString }}
```
👍 Reviewed on the front page, near the bottom.

jpluscplusm reviewed (approved):
LGTM!

jpluscplusm commented:
Imported as https://cuelang.org/cl/1203530.
